cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -124,6 +125,15 @@ class LeadImpl extends ServerImpl with Lead with Logging {
         conf.set(key, v)
       })
 
+      /**
+       * If kryoserializer is being used then register snappyclasses for kryoserializer.
+       * The snappyclasses will not get registered if user is creating its own spark context.
+       * In that case user has to specify the snappy kryo registrator externally
+       */
+      if (conf.contains("spark.serializer") && conf.get("spark.serializer").equals("org.apache.spark.serializer.KryoSerializer")) {
+        conf.set("spark.kryo.registrator", getSnappyKryoRegistrarClass())
```

Comment:
Why user can not create a Spark conf ? I thought lead node has options for that. There must be.
It will be too much for user to extend our registrator. We will go by this route if no option left for us. Can we find out why we need to register our classes in kryo ? If ser/deser is in our control why we can't emulate SparkSQlSerializer, which just extends KryoSerializer with its own addition..  

For CachedBatch serialization we can surely extend SparkSQLSerializer. What's the deal with AQP related classes . Can we please find out ?